src/components/powerInfoPage/table.tsx

```diff
@@ -0,0 +1,303 @@
+          </div>
+          <div className={classNames('powerinfo-table-list-item-time')}>
+            {d.time}
+          </div>
+          <div className={classNames('powerinfo-table-list-item-value')}>
+            {d.power_display}
+          </div>
+          <div
+            className={classNames('powerinfo-table-list-item-type')}
+            style={d.power_display >= 0 ? redText : greenText}
+          >
+            {dataType(d.data_type)}
+          </div>
+          <div className={classNames('powerinfo-table-list-item-url')}>
+            <a href={d.address}>&#60;URL&#62;</a>
+          </div>
+          <div className={classNames('powerinfo-table-list-item-infobox')}>
+            <InfoBox
+              date={d.date}
+              type={d.data_type}
+              time={d.time}
+              value={d.power_display}
+              url={d.address}
+            />
+          </div>
+        </div>
+        <hr className={classNames('powerinfo-table-list-item-line')} />
+      </div>
+    );
+  });
+
+  const selectChangeHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPage(1);
+    setPerPage(event.target.value);
+  };
+
+  const clickFirstPageHandler = () => {
+    setPage(1);
+  };
+
+  const clickPrevPageHandler = () => {
+    if (page > 1) setPage(page - 1);
+    else setPage(1);
+  };
+
+  const clickNextPageHandler = () => {
+    const lastPage =
+      apiData.totalCount % parseInt(per_page, 10) === 0
+        ? apiData.totalCount / parseInt(per_page, 10)
+        : Math.floor(apiData.totalCount / parseInt(per_page, 10)) + 1;
+    if (page < lastPage) setPage(page + 1);
+    else setPage(lastPage);
+  };
+
+  const clickLastPageHandler = () => {
+    const lastPage =
+      apiData.totalCount % parseInt(per_page, 10) === 0
+        ? apiData.totalCount / parseInt(per_page, 10)
+        : Math.floor(apiData.totalCount / parseInt(per_page, 10)) + 1;
+    setPage(lastPage);
+  };
+
+  const [prevDisabled, setPrevDisabled] = useState(true);
+  const [nextDisabled, setNextDisabled] = useState(false);
+
+  useEffect(() => {
+    const lastPage =
+      apiData.totalCount % parseInt(per_page, 10) === 0
+        ? apiData.totalCount / parseInt(per_page, 10)
+        : Math.floor(apiData.totalCount / parseInt(per_page, 10)) + 1;
+    if (page === 1 && lastPage === 1) {
+      setPrevDisabled(true);
+      setNextDisabled(true);
+    } else if (page === 1) {
+      setPrevDisabled(true);
+      setNextDisabled(false);
+    } else if (page === lastPage) {
+      setPrevDisabled(false);
+      setNextDisabled(true);
+    } else {
+      setPrevDisabled(false);
+      setNextDisabled(false);
+    }
+  }, [page]);
+
+  useEffect(() => {
+    (async () => {
+      await fetchApiData();
+    })();
+    // eslint-disable-next-line @typescript-eslint/camelcase
+  }, [correctDate, per_page, page]);
+
+  const [timeText, setTimeText] = useState('紀錄時間');
+  const [typeText, setTypeText] = useState('用產電種類');
```

Comment:
Please move these two line to the top of the functional component with other useState variables.

Reply:
Done. Thanks!